src/components/DatePicker/index.js

```diff
@@ -48,36 +53,27 @@ function DatePicker({containerStyles, defaultValue, disabled, errorText, inputID
     const styles = useThemeStyles();
     const [selectedDate, setSelectedDate] = useState(value || defaultValue || undefined);
 
-    useEffect(() => {
-        if (selectedDate === value || _.isUndefined(value)) {
-            return;
-        }
-        setSelectedDate(value);
-    }, [selectedDate, value]);
-
-    useEffect(() => {
+    const onSelected = (newValue) => {
         if (_.isFunction(onTouched)) {
             onTouched();
         }
         if (_.isFunction(onInputChange)) {
-            onInputChange(selectedDate);
+            onInputChange(newValue);
         }
-        // To keep behavior from class component state update callback, we want to run effect only when the selected date is changed.
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [selectedDate]);
+        setSelectedDate(newValue);
+    };
```

Comment:
Any specific reason on why we are calling `onInputChange` too early? Calling it in the `useEffect` (as it was) looks better since it's a side effect.


```js
    useEffect(() => {
        if (_.isFunction(onTouched)) {
            onTouched();
        }
        if (_.isFunction(onInputChange)) {
            onInputChange(selectedDate);
        }
    }, [selectedDate]);

```
```js
                    onSelected={setSelectedDate}
```

Reply:
I disagree that it's called too early. Currently its behaviour is similar to `onValueChange` from `FormProvider` - it's a callback, which is fired when we are certain, that the state will be changed and we know the new state. Additionally it seems like event-specific logic inside effect, and that's a [React anti-patern].(https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers)

Reply:
Makes sense!